### ISSUE TITLE
datalad get fails

### ISSUE THREAD

**Issue by oesteban** [templateflow/tpl-NKI#1]
I thought I had reported this before??

```
tpl-NKI$ datalad get *
Total (0 ok, 15 failed out of 15): 70.0MB [00:06, 22.0MB/s]                                                                                                                                                     [WARNING] Running get resulted in stderr output:   verification of content failed                                                                                                                                 
  verification of content failed                                                                                                                                                                                 
  verification of content failed.nii.gz:   0%|                                                                                                                                       | 0.00/3.93M [00:00<?, ?B/s]
  verification of content failednii.gz:   0%|                                                                                                                                         | 0.00/190k [00:00<?, ?B/s]
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
  verification of content failed
git-annex: get: 15 failed
 
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-BrainCerebellumExtraction_mask.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-BrainCerebellumExtraction_mask.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-BrainCerebellumRegistration_mask.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-BrainCerebellumRegistration_mask.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-malf_dseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-malf_dseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-6_dseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-6_dseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-brain_mask.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-brain_mask.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-brain_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-brain_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-BS_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-BS_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-brainNoCerebellum_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-brainNoCerebellum_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-brain_T1w.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_desc-brain_T1w.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-SCGM_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-SCGM_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-CBM_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-CBM_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-CGM_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-CGM_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-CSF_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-CSF_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-WM_probseg.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_label-WM_probseg.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_T1w.nii.gz)] 
get(error): /home/oesteban/datalad/templateflow/tpl-NKI/tpl-NKI_res-01_T1w.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
action summary:
  get (error: 15, notneeded: 6)
```

**Comment by yarikoptic** [comment]
Smells like content pointed by the url changed? I will check when get to the laptop

**Comment by yarikoptic** [comment]
yeap, either content changed or incorrect url was "registered" to those files -- since they used "addurl --fast" mode, their key is defined by URL, not checksum but still includes the size of the file, which is now seems to be different. Here is a full protocol of my quick exploration:

```shell
$> datalad get tpl-NKI_res-01_T1w.nii.gz 
[WARNING] Running get resulted in stderr output:   verification of content failed                                                                                                              
git-annex: get: 1 failed
 
[ERROR  ] from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin) [get(/home/yoh/proj/templateflow/templateflow/tpl-NKI/tpl-NKI_res-01_T1w.nii.gz)] 
get(error): /home/yoh/proj/templateflow/templateflow/tpl-NKI/tpl-NKI_res-01_T1w.nii.gz (file) [from web...; Unable to access these remotes: web; Try making some of these repositories available:; 	00000000-0000-0000-0000-000000000001 -- web; ; (Note that these git remotes have annex-ignore set: origin)]
datalad get tpl-NKI_res-01_T1w.nii.gz  4.58s user 0.45s system 53% cpu 9.364 total
1 11411 ->1.....................................:Fri 21 Dec 2018 11:13:50 PM EST:.
(git-annex)hopa:~/proj/templateflow/templateflow/tpl-NKI[master]
$> git annex whereis tpl-NKI_res-01_T1w.nii.gz
whereis tpl-NKI_res-01_T1w.nii.gz (1 copy) 
  	00000000-0000-0000-0000-000000000001 -- web

  web: https://files.osf.io/v1/resources/ue5gx/providers/osfstorage/5bc6481853cec4001baea563
ok
1 11412.....................................:Fri 21 Dec 2018 11:13:55 PM EST:.
(git-annex)hopa:~/proj/templateflow/templateflow/tpl-NKI[master]
$> ls -ld tpl-NKI_res-01_T1w.nii.gz
lrwxrwxrwx 1 yoh yoh 186 Dec 21 23:09 tpl-NKI_res-01_T1w.nii.gz -> .git/annex/objects/2V/87/URL-s43131574--https&c%%files.osf.io%v1%resourc-389c568079306fb3ac5f6713f700e8c2/URL-s43131574--https&c%%files.osf.io%v1%resourc-389c568079306fb3ac5f6713f700e8c2
1 11413.....................................:Fri 21 Dec 2018 11:14:01 PM EST:.
(git-annex)hopa:~/proj/templateflow/templateflow/tpl-NKI[master]
$> wget https://files.osf.io/v1/resources/ue5gx/providers/osfstorage/5bc6481853cec4001baea563
--2018-12-21 23:14:14--  https://files.osf.io/v1/resources/ue5gx/providers/osfstorage/5bc6481853cec4001baea563
Resolving files.osf.io (files.osf.io)... 35.186.214.196
Connecting to files.osf.io (files.osf.io)|35.186.214.196|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://storage.googleapis.com/cos-osf-prod-files-us/d556bdf042a6b8a6bc413f0f68d3c8399a7718db176cd1ca4df13907e4aecc4d?Expires=1545452115&GoogleAccessId=files-us%40cos-osf-prod.iam.gserviceaccount.com&Signature=IhfEEAzM1f9y7SWr6DA%2BGApJnuSfKmEAtcmeSqzIQtmqDD5L4J%2FPU1cIDeUoGP54vxU63dA3oCBzpIj3TmmzeNjRQ46rvdNcURh%2FO4m9WR64f801RX0IHuTFP8rrSEL%2FkwvD7dWoGOlDa7p%2F7FOsA9MBc2MNpY3yY2lkU9yNGrrw9Gc7q%2BmLTznOe%2BctXk8mZWNSukcrsxUbJk3YUOM0NBxGY4UJ0IGD06TePZugFg4QJg9w%2F6u7RTIYTr62pypubJVgt8eaodFlg5kuc88ycVTHwQ%2BPNKCWOCjqwI6mAmeHtFFkErZefECJN13GlppUAImcUKC3bWrA07ilxLFHFA%3D%3D&response-content-disposition=attachment%3B%20filename%3D%22tpl-NKI_res-01_T1w.nii.gz%22%3B%20filename%2A%3DUTF-8%27%27tpl-NKI_res-01_T1w.nii.gz [following]
--2018-12-21 23:14:15--  https://storage.googleapis.com/cos-osf-prod-files-us/d556bdf042a6b8a6bc413f0f68d3c8399a7718db176cd1ca4df13907e4aecc4d?Expires=1545452115&GoogleAccessId=files-us%40cos-osf-prod.iam.gserviceaccount.com&Signature=IhfEEAzM1f9y7SWr6DA%2BGApJnuSfKmEAtcmeSqzIQtmqDD5L4J%2FPU1cIDeUoGP54vxU63dA3oCBzpIj3TmmzeNjRQ46rvdNcURh%2FO4m9WR64f801RX0IHuTFP8rrSEL%2FkwvD7dWoGOlDa7p%2F7FOsA9MBc2MNpY3yY2lkU9yNGrrw9Gc7q%2BmLTznOe%2BctXk8mZWNSukcrsxUbJk3YUOM0NBxGY4UJ0IGD06TePZugFg4QJg9w%2F6u7RTIYTr62pypubJVgt8eaodFlg5kuc88ycVTHwQ%2BPNKCWOCjqwI6mAmeHtFFkErZefECJN13GlppUAImcUKC3bWrA07ilxLFHFA%3D%3D&response-content-disposition=attachment%3B%20filename%3D%22tpl-NKI_res-01_T1w.nii.gz%22%3B%20filename%2A%3DUTF-8%27%27tpl-NKI_res-01_T1w.nii.gz
Resolving storage.googleapis.com (storage.googleapis.com)... 172.217.3.112, 2607:f8b0:4004:804::2010
Connecting to storage.googleapis.com (storage.googleapis.com)|172.217.3.112|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 43131570 (41M) [application/octet-stream]
Saving to: ‘5bc6481853cec4001baea563’

5bc6481853cec4001baea563                        100%[======================================================================================================>]  41.13M  6.44MB/s    in 6.5s    

2018-12-21 23:14:22 (6.35 MB/s) - ‘5bc6481853cec4001baea563’ saved [43131570/43131570]

1 11414.....................................:Fri 21 Dec 2018 11:14:22 PM EST:.
(git-annex)hopa:~/proj/templateflow/templateflow/tpl-NKI[master]
$> du -scb 5bc6481853cec4001baea563 
43131570	5bc6481853cec4001baea563
43131570	total
```

git log shows that it never was of that size, but also that it jumped from MD5E backend to url backend... did you use `--fast` option on the 2nd run of addurls?
```shell
(git-annex)hopa:~/proj/templateflow/templateflow/tpl-NKI[master]
$> git log -p tpl-NKI_res-01_T1w.nii.gz
commit 51037c36d2e3fca762c0a892c3d92b44d0a510bf
Author: rciric <rc042010@mymail.pomona.edu>
Date:   Sat Dec 15 05:14:35 2018 -0800

    [DATALAD] add files from URLs
    
    url_file='/tmp/tpl-NKI.csv'
    url_format='{url}'
    filename_format='{path}'

diff --git a/tpl-NKI_res-01_T1w.nii.gz b/tpl-NKI_res-01_T1w.nii.gz
new file mode 120000
index 0000000..b486c79
--- /dev/null
+++ b/tpl-NKI_res-01_T1w.nii.gz
@@ -0,0 +1 @@
+.git/annex/objects/2V/87/URL-s43131574--https&c%%files.osf.io%v1%resourc-389c568079306fb3ac5f6713f700e8c2/URL-s43131574--https&c%%files.osf.io%v1%resourc-389c568079306fb3ac5f6713f700e8c2
\ No newline at end of file

commit ab49b39e29a74fb2996eeba11a1a6b9f97ffb836
Author: rciric <rc042010@mymail.pomona.edu>
Date:   Sat Dec 15 05:13:29 2018 -0800

    [DATALAD] removed content

diff --git a/tpl-NKI_res-01_T1w.nii.gz b/tpl-NKI_res-01_T1w.nii.gz
deleted file mode 120000
index 8b86169..0000000
--- a/tpl-NKI_res-01_T1w.nii.gz
+++ /dev/null
@@ -1 +0,0 @@
-.git/annex/objects/Mk/xm/MD5E-s43131574--f938471d3d40cdaf83e9389415e52060.nii.gz/MD5E-s43131574--f938471d3d40cdaf83e9389415e52060.nii.gz
\ No newline at end of file

commit df018440c4eb5814003a6ded6af60b599facb5cb
Author: oesteban <code@oscaresteban.es>
Date:   Thu Oct 18 23:41:18 2018 -0700

    [DATALAD] add files from URLs
    
    url_file='../urlslin.csv'
    url_format='{url}'
    filename_format='{filename}'

diff --git a/tpl-NKI_res-01_T1w.nii.gz b/tpl-NKI_res-01_T1w.nii.gz
new file mode 120000
index 0000000..8b86169
--- /dev/null
+++ b/tpl-NKI_res-01_T1w.nii.gz
@@ -0,0 +1 @@
+.git/annex/objects/Mk/xm/MD5E-s43131574--f938471d3d40cdaf83e9389415e52060.nii.gz/MD5E-s43131574--f938471d3d40cdaf83e9389415e52060.nii.gz
\ No newline at end of file
```

**Comment by rciric** [comment]
Hello @yarikoptic , I did indeed use the `--fast` option on `addurls` (line 17 [here](https://github.com/rciric/datalad-osf-utils/blob/master/utils.py#L17)). (And Oscar did report this previously -- I deleted it by accident when I was trying to remove a dirty fork; sorry about that.)

I noticed that the OSF datalad [interface](https://github.com/rciric/datalad-osf-utils/blob/master/utils.py) I cobbled together results in failed verification when I use `addurls` with modified (version 2 or later) files on OSF. (NKI files in particular are version 2 since the originals were RAS+ and I had to reorient them -- perhaps the file size key is still referencing the original version of the file.)

If I use version 1 files (i.e., delete the files that are currently there and then reupload -- did this in an OSF fork), it works fine (see [here](https://github.com/rciric/tpl-NKI)), but this is not ideal since we lose the version control utility that OSF gives us in the first place.

Looking at the NKI [metadata on OSF](https://files.osf.io/v1/resources/ue5gx/providers/osfstorage/5bc647d653cec4001baea4bb/), there doesn't seem to be an obvious change that would make the interface work for files that are version 2 or later, though.

I can try setting `fast=False` first and seeing if that helps any.
